modules/monitoring-creating-scrape-sample-alerts.adoc

```diff
@@ -0,0 +1,79 @@
+
+* You have access to the cluster as a user with the `cluster-admin` role, or as a user with the `user-workload-monitoring-config-edit` role in the `openshift-user-workload-monitoring` project.
+* You have enabled monitoring for user-defined projects.
+* You have created the `user-workload-monitoring-config` ConfigMap object.
+* You have limited the number of samples that can be accepted per target scrape in user-defined projects, by using `enforcedSampleLimit`.
+* You have installed the OpenShift CLI (`oc`).
+
+.Procedure
+
+. Create a YAML file with alerts that inform you when the targets are down and when the enforced sample limit is approaching. The file in this example is called `monitoring-stack-alerts.yaml`:
++
+[source,yaml]
+----
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  labels:
+    prometheus: k8s
+    role: alert-rules
+  name: monitoring-stack-alerts <1>
+  namespace: ns1 <2>
+spec:
+  groups:
+  - name: general.rules
+    rules:
+    - alert: TargetDown <3>
+      annotations:
+        message: '{{ printf "%.4g" $value }}% of the {{ $labels.job }}/{{ $labels.service
+          }} targets in {{ $labels.namespace }} namespace are down.' <4>
+      expr: 100 * (count(up == 0) BY (job, namespace, service) / count(up) BY (job,
+        namespace, service)) > 10
+      for: 10m <5>
+      labels:
+        severity: warning <6>
+    - alert: ApproachingEnforcedSamplesLimit <7>
+      annotations:
+        message: '{{ $labels.container }} container of the {{ $labels.pod }} pod in the {{ $labels.namespace }} namespace consumes {{ $value | humanizePercentage }} of the samples limit budget.' <8>
+      expr: scrape_samples_scraped/50000 > 0.8 <9>
+      for: 10m <10>
+      labels:
+        severity: warning <11>
+----
+<1> Defines the name of the alerting rule.
+<2> Specifies the user-defined project where the alerting rule will be deployed.
+<3> The `TargetDown` alert will fire if the target cannot be scraped or is not available for the `for` duration.
+<4> The message that will be output when the `TargetDown` alert fires.
+<5> The `TargetDown` alert's conditions must be true for this duration before the alert is fired.
+<6> Defines the `TargetDown` alert's severity.
+<7> The `ApproachingEnforcedSamplesLimit` alert will fire when the defined scrape sample threshold is reached or exceeded for the specified `for` duration.
+<8> The message that will be output when the `ApproachingEnforcedSamplesLimit` alert fires.
+<9> The `ApproachingEnforcedSamplesLimit` alert's threshold. In this example the alert will fire when the number of samples per target scrape has reached or exceeded 80% of the enforced sample limit of `50000`. The `for` duration must also have passed before the alert will fire.
++
+[NOTE]
+====
+The number in the expression `scrape_samples_scraped/<number> > <threshold>`, `<number>` must match the `enforcedSampleLimit` value defined in the `user-workload-monitoring-config` ConfigMap.
```

Comment:
Same here re: admonition box breaking the callout list numbering.

Reply:
Thank you. I have now moved the note.

Reply:
On review I have added the content from the note into call out 9. I think it needs to be read in context to that specific call out directly.